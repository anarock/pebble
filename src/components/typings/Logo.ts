export interface LogoProps {
  height?: number;
  width?: number;
  color?: string;
}
