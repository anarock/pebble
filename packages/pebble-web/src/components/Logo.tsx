import * as React from "react";
import { css } from "emotion";
import { LogoProps } from "./typings/Logo";

const Logo: React.FunctionComponent<LogoProps> = ({
  height = 40,
  color = "#000000"
}) => {
  const style = css({
    fill: color
  });

  return (
    <svg
      xmlnsXlink="http://www.w3.org/1999/xlink"
      height={height}
      viewBox="0 0 413.31 92.91"
    >
      <defs>
        <clipPath id="clip-path" transform="translate(-57.88 -61.39)">
          <rect className="cls-1" width="529.46" height="215.51" />
        </clipPath>
      </defs>
      <g id="Layer_2" data-name="Layer 2">
        <g className="cls-2" id="Layer_1-2" data-name="Layer 1">
          <polygon
            className={style}
            points="7.31 91.39 3.84 82.98 2.62 82.98 6.81 92.81 7.79 92.81 11.98 82.98 10.8 82.98 7.31 91.39"
          />
          <path
            className={style}
            d="M89.05 144.31l-4.45 9.82h1.13l1.16-2.57h5.31l1.14 2.57h1.18l-4.44-9.82zm-1.73 6.24l2.23-5 2.22 5z"
            transform="translate(-57.88 -61.39)"
          />
          <polygon
            className={style}
            points="54.71 82.98 53.61 82.98 53.61 92.74 60.2 92.74 60.2 91.72 54.71 91.72 54.71 82.98"
          />
          <path
            className={style}
            d="M141.75 150.05c0 2.13-1.14 3.22-2.94 3.22s-3-1.18-3-3.29v-5.6h-1.1v5.67c0 2.76 1.64 4.24 4.05 4.24s4.07-1.48 4.07-4.32v-5.59h-1.1z"
            transform="translate(-57.88 -61.39)"
          />
          <polygon
            className={style}
            points="103.72 88.32 109.04 88.32 109.04 87.32 103.72 87.32 103.72 83.99 109.67 83.99 109.67 82.98 102.62 82.98 102.62 92.74 109.74 92.74 109.74 91.74 103.72 91.74 103.72 88.32"
          />
          <path
            className={style}
            d="M186.15 148.73c-2.14-.46-2.65-1-2.65-1.9 0-.87.81-1.57 2.08-1.57a4.3 4.3 0 0 1 2.82 1.06l.64-.85a5.07 5.07 0 0 0-3.43-1.2c-1.85 0-3.21 1.13-3.21 2.66 0 1.6 1 2.33 3.28 2.81 2 .44 2.54.95 2.54 1.86 0 1-.87 1.65-2.18 1.65a4.63 4.63 0 0 1-3.34-1.38l-.69.81a5.69 5.69 0 0 0 4 1.56c1.94 0 3.33-1.08 3.33-2.76 0-1.49-1-2.29-3.18-2.76m44.24-4.51a5 5 0 0 0-5 5 5 5 0 1 0 10 0 4.9 4.9 0 0 0-5-5m3.82 5.07a3.83 3.83 0 0 1-3.82 4 3.88 3.88 0 0 1-3.85-4 3.83 3.83 0 0 1 3.82-4 3.88 3.88 0 0 1 3.85 4z"
            transform="translate(-57.88 -61.39)"
          />
          <polygon
            className={style}
            points="198.14 91.39 194.67 82.98 193.44 82.98 197.64 92.81 198.61 92.81 202.81 82.98 201.62 82.98 198.14 91.39"
          />
          <polygon
            className={style}
            points="220.87 88.32 226.2 88.32 226.2 87.32 220.87 87.32 220.87 83.99 226.82 83.99 226.82 82.98 219.77 82.98 219.77 92.74 226.9 92.74 226.9 91.74 220.87 91.74 220.87 88.32"
          />
          <path
            className={style}
            d="M309.76 147.31a2.73 2.73 0 0 0-.75-1.93 3.88 3.88 0 0 0-2.78-1H302v9.75h1.1v-3.79h2.78l2.83 3.79h1.35l-3-4a2.86 2.86 0 0 0 2.65-2.84m-6.62 2v-3.89h3c1.57 0 2.49.72 2.49 1.92 0 1.25-1 2-2.51 2z"
            transform="translate(-57.88 -61.39)"
          />
          <polygon
            className={style}
            points="292.36 91.39 288.89 82.98 287.66 82.98 291.86 92.81 292.83 92.81 297.03 82.98 295.84 82.98 292.36 91.39"
          />
          <path
            className={style}
            d="M374.09 144.31l-4.44 9.82h1.13l1.15-2.57h5.31l1.15 2.57h1.18l-4.45-9.82zm-1.73 6.24l2.23-5 2.22 5z"
            transform="translate(-57.88 -61.39)"
          />
          <polygon
            className={style}
            points="339.75 82.98 338.65 82.98 338.65 92.74 345.24 92.74 345.24 91.72 339.75 91.72 339.75 82.98"
          />
          <path
            className={style}
            d="M426.8 150.05c0 2.13-1.14 3.22-2.94 3.22s-3-1.18-3-3.29v-5.6h-1.11v5.67c0 2.76 1.65 4.24 4.06 4.24s4.07-1.48 4.07-4.32v-5.59h-1.1z"
            transform="translate(-57.88 -61.39)"
          />
          <polygon
            className={style}
            points="388.76 88.32 394.09 88.32 394.09 87.32 388.76 87.32 388.76 83.99 394.72 83.99 394.72 82.98 387.66 82.98 387.66 92.74 394.79 92.74 394.79 91.74 388.76 91.74 388.76 88.32"
          />
          <rect className={style} y="76.12" width="399.11" height="0.74" />
          <polygon
            className={style}
            points="72.54 33.07 117.31 71.74 117.31 11.02 107.61 11.03 107.61 51.1 63.42 11.18 63.42 71.06 72.54 71.06 72.54 33.07"
          />
          <path
            className={style}
            d="M193.23 120.52h27.68l5.81 12h10.95l-30.41-62L177 132.5h10.44zm4.12-9.93l9.33-19.19 9.32 19.19zm-122.42 9.93h27.68l5.82 12h10.95L89 70.47l-30.29 62h10.4zm4.12-9.93l9.33-19.19 9.33 19.19zM248.39 120h2.89v13h2.81v-15.81h-8.5v13.05H242v-18.56l7.22-6.61 27.3 27.66h13.89l-26.89-26.54c5.63-.34 21.73-2.47 21.73-15.85 0-11.29-8.31-18-22.23-18h-23.71V133h9.09zm14.13-22.2H250V82.33h12.3c7.59 0 11.94 2.59 11.94 7.1.04 7.26-7.33 8.34-11.72 8.34zm55.55 34.72c18.06 0 30.67-12.37 30.67-30.09A30 30 0 0 0 340.28 81c-5.55-5.55-13.44-8.6-22.21-8.6-17.84 0-30.78 12.65-30.78 30.08s12.94 30.09 30.78 30.09m-21.41-30.1c0-13.3 8.81-22.6 21.43-22.6s21.27 9.3 21.27 22.6-8.75 22.74-21.27 22.74-21.43-9.4-21.43-22.79zm84.28 30.26c10.83 0 19-4.43 24.42-13.17l.25-.4-6.25-5.06-.31.57c-4.47 8.19-11.83 9.91-17.21 9.91-12.38 0-20.69-8.94-20.69-22.24 0-12.2 8.59-21.4 20-21.4 7.29 0 13.67 2.68 16.24 6.82l.37.6 5.77-6.15-.27-.36c-4.64-6.25-12-9.29-22.53-9.29-17.09 0-30 12.83-30 29.85 0 17.85 12.42 30.32 30.2 30.32"
            transform="translate(-57.88 -61.39)"
          />
          <polygon
            className={style}
            points="360.13 46.74 362.55 44.05 386.41 70.99 398.53 70.99 369.14 37.48 394.27 11.15 382.73 11.15 360.13 34.58 360.13 11.02 349.68 11.02 349.68 71.02 360.13 71.02 360.13 46.74"
          />
          <path
            className={style}
            d="M461.59 68.58h1.6a2.58 2.58 0 0 0 1.57-.34 1.08 1.08 0 0 0 .42-.9 1.1 1.1 0 0 0-.2-.64 1.28 1.28 0 0 0-.56-.43 4.3 4.3 0 0 0-1.33-.14h-1.5zm-1.31 4.6V65h2.8a6.88 6.88 0 0 1 2.09.23 2 2 0 0 1 1 .79 2.1 2.1 0 0 1 .38 1.19 2.17 2.17 0 0 1-.64 1.56 2.53 2.53 0 0 1-1.7.74 2.27 2.27 0 0 1 .69.43 9.29 9.29 0 0 1 1.21 1.62l1 1.6h-1.6l-.73-1.29a6.64 6.64 0 0 0-1.39-1.9 1.67 1.67 0 0 0-1.05-.27h-.78v3.46zm3.33-10.53a6.55 6.55 0 0 0-3.12.82 5.89 5.89 0 0 0-2.37 2.35 6.31 6.31 0 0 0 0 6.31 5.9 5.9 0 0 0 2.35 2.35 6.33 6.33 0 0 0 6.3 0 5.88 5.88 0 0 0 2.34-2.35 6.35 6.35 0 0 0 .84-3.14 6.47 6.47 0 0 0-.85-3.17 5.89 5.89 0 0 0-2.37-2.35 6.57 6.57 0 0 0-3.12-.82zm0-1.26a7.72 7.72 0 0 1 3.73 1 7 7 0 0 1 2.84 2.8 7.82 7.82 0 0 1 1 3.81 7.63 7.63 0 0 1-1 3.77 7 7 0 0 1-2.8 2.81 7.58 7.58 0 0 1-7.56 0 7 7 0 0 1-2.82-2.82 7.64 7.64 0 0 1-1-3.76 7.72 7.72 0 0 1 1-3.81 7 7 0 0 1 2.84-2.8 7.76 7.76 0 0 1 3.77-1z"
            transform="translate(-57.88 -61.39)"
          />
        </g>
      </g>
    </svg>
  );
};

export default Logo;
