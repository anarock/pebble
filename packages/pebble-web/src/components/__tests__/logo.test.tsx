import * as React from "react";
import renderer from "react-test-renderer";
import Logo from "../Logo";

describe("Logo", () => {
  test("snapshot", () => {
    const logo = renderer.create(<Logo />);

    const tree = logo.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
