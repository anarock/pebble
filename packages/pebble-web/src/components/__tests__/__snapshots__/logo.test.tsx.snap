// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo snapshot 1`] = `
<svg
  height={40}
  id="Layer_1"
  style={
    Object {
      "fill": "#000000",
    }
  }
  version="1.1"
  viewBox="0 0 655.35 175.08"
  xmlSpace="preserve"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <path
    className="st0"
    d="M168.92 65.96l-66.1-64.85v100.61h17.39V40.78l66.11 64.56V4.73h-17.4v61.23zm446.77-18.51L653.4 4.73h-20.6l-41.04 46.48V4.73h-17.4v96.99h17.4V73.34L604 59.84l30.19 41.88h20.33l-38.83-54.27zM422.66 2.92c-27.78 0-50.31 22.53-50.31 50.31s22.52 50.3 50.31 50.3c27.78 0 50.3-22.52 50.3-50.3s-22.52-50.31-50.3-50.31zm0 84.84c-19.05 0-34.54-15.49-34.54-34.53s15.49-34.54 34.54-34.54c19.04 0 34.54 15.49 34.54 34.54 0 19.04-15.5 34.53-34.54 34.53zM0 101.72h18.38L27 82.93h41.75l8.63 18.79h18.51L48.02 0 0 101.72zM33.55 67.9l14.32-31.58L62.21 67.9H33.55zm159.61 33.82h18.37l8.62-18.79h41.75l8.62 18.79h18.51L241.16 0l-48 101.72zm33.53-33.82l14.33-31.58 14.33 31.58h-28.66zm138.43-32.42c0-8.76-3.07-16-9.19-21.84-6.12-5.99-13.5-8.91-22.26-8.91h-37.71v96.99h17.4V67.21h17.25l20.73 34.51h19.62l-23.51-38.68c10.57-4.46 17.67-14.89 17.67-27.56zm-33.25 16.15h-18.51V20.46h19.9c8.34 0 14.6 6.4 14.6 15.31 0 9.32-6.54 15.86-15.99 15.86zm197.61-32.94c9.22 0 17.9 3.59 24.42 10.12l11.15-11.15c-9.82-9.83-22.7-14.74-35.57-14.74-12.87 0-25.75 4.91-35.56 14.74-19.65 19.64-19.65 51.49 0 71.14 9.82 9.82 22.69 14.73 35.56 14.73 12.87 0 25.75-4.91 35.57-14.73L553.9 77.65c-6.52 6.52-15.2 10.11-24.42 10.11-9.23 0-17.89-3.59-24.42-10.11-13.46-13.46-13.46-35.37 0-48.84 6.53-6.52 15.2-10.12 24.42-10.12z"
  />
  <rect
    className="st1"
    height="4.26"
    id="XMLID_9_"
    width="654.52"
    y="123.84"
  />
  <path
    className="st0"
    d="M90.05 150.11h6.02l6.48 17.46 6.48-17.46h5.88l-10.02 24.97h-4.81l-10.03-24.97zm39.59-.17h5.03l10.63 24.97h-5.7l-2.28-5.56h-10.47l-2.27 5.56h-5.56l10.62-24.97zm5.75 14.59l-3.3-8.04-3.29 8.04h6.59zM154 150.11h5.46v19.83h12.36v4.96H154v-24.79zm36.79 25.18c-3.33 0-5.95-.92-7.86-2.76-1.91-1.83-2.87-4.59-2.87-8.25v-14.16h5.46v14.02c0 2.03.46 3.56 1.41 4.58.94 1.03 2.26 1.54 3.93 1.54 1.68 0 2.98-.49 3.93-1.49.95-.99 1.42-2.48 1.42-4.46v-14.2h5.46v13.99c0 1.88-.26 3.53-.77 4.92-.51 1.39-1.23 2.55-2.17 3.49-.95.93-2.08 1.63-3.44 2.08-1.35.47-2.85.7-4.5.7zm21.83-25.18h18.66v4.85H218v5.03h11.7v4.85H218v5.2h13.46v4.86h-18.84v-24.79zm37.74 25.15c-1.89 0-3.75-.33-5.57-.98-1.82-.65-3.47-1.64-4.94-2.99l3.21-3.86c1.14.92 2.3 1.65 3.49 2.19 1.2.54 2.51.82 3.91.82 1.13 0 2.01-.21 2.64-.62.63-.41.94-.99.94-1.72v-.07c0-.36-.07-.67-.19-.94-.14-.27-.39-.53-.75-.77-.36-.23-.87-.47-1.52-.71s-1.49-.48-2.53-.74c-1.26-.31-2.38-.64-3.4-1.03-1.01-.38-1.88-.84-2.59-1.4a5.85 5.85 0 0 1-1.65-2.07c-.38-.82-.58-1.86-.58-3.08v-.07c0-1.13.21-2.16.63-3.07.42-.91 1.01-1.7 1.78-2.35.76-.66 1.67-1.17 2.73-1.52 1.06-.35 2.23-.54 3.5-.54 1.83 0 3.5.28 5.02.82 1.52.54 2.92 1.32 4.19 2.34l-2.83 4.11c-1.1-.76-2.2-1.35-3.26-1.79-1.06-.44-2.12-.66-3.19-.66-1.06 0-1.85.21-2.38.62-.53.41-.8.93-.8 1.54v.07c0 .4.08.75.23 1.04.15.29.43.56.84.79.39.24.94.46 1.64.67.7.21 1.57.47 2.63.75 1.26.33 2.37.7 3.35 1.11.98.41 1.81.91 2.48 1.49.68.58 1.18 1.25 1.52 2.04.34.78.52 1.71.52 2.8v.07c0 1.22-.23 2.32-.67 3.27a6.71 6.71 0 0 1-1.86 2.41c-.81.65-1.76 1.14-2.87 1.48-1.11.37-2.33.55-3.67.55zm46.42.07c-1.92 0-3.68-.34-5.28-1.01-1.61-.67-3-1.58-4.16-2.72-1.17-1.15-2.08-2.5-2.73-4.04-.65-1.55-.98-3.2-.98-4.98v-.07c0-1.77.32-3.42.99-4.97a12.64 12.64 0 0 1 2.74-4.07c1.16-1.17 2.57-2.09 4.18-2.77 1.62-.67 3.39-1.01 5.3-1.01 1.91 0 3.67.34 5.28 1.01 1.61.68 2.99 1.59 4.16 2.73 1.16 1.14 2.07 2.49 2.73 4.04.64 1.54.97 3.21.97 4.98v.07c0 1.77-.33 3.43-.99 4.98-.67 1.55-1.57 2.9-2.74 4.07-1.17 1.17-2.56 2.08-4.18 2.76-1.61.66-3.37 1-5.29 1zm.08-5.03c1.08 0 2.09-.2 3.01-.6.92-.4 1.71-.95 2.35-1.66.65-.71 1.16-1.53 1.53-2.47.36-.93.54-1.93.54-2.99v-.07c0-1.06-.18-2.06-.54-3.01-.36-.94-.89-1.77-1.56-2.47a7.49 7.49 0 0 0-2.39-1.68c-.91-.41-1.93-.62-3.01-.62-1.11 0-2.12.21-3.03.6-.91.4-1.69.96-2.34 1.66a7.76 7.76 0 0 0-1.52 2.46c-.37.94-.55 1.94-.55 3v.07c0 1.06.18 2.07.55 3.02.36.94.88 1.76 1.55 2.47.67.71 1.47 1.27 2.38 1.68.91.4 1.91.61 3.03.61zm19.48-20.19h6.02l6.48 17.46 6.48-17.46h5.88l-10.02 24.97h-4.81l-10.03-24.97zm33.54 0h18.66v4.85h-13.28v5.03h11.68v4.85h-11.68v5.2h13.46v4.86h-18.84v-24.79zm29.16 0h11.34c3.13 0 5.54.83 7.22 2.51 1.42 1.42 2.12 3.31 2.12 5.67v.07c0 2.01-.49 3.65-1.47 4.91-.98 1.26-2.26 2.19-3.84 2.77l6.05 8.86h-6.37l-5.31-7.94h-4.3v7.94h-5.45v-24.79zm10.98 12.04c1.35 0 2.39-.32 3.1-.95.72-.64 1.08-1.48 1.08-2.54v-.07c0-1.17-.37-2.06-1.13-2.64-.75-.59-1.8-.88-3.15-.88h-5.42v7.08h5.52zm33.05-12.04h6.02l6.48 17.46 6.48-17.46h5.87l-10.02 24.97h-4.81l-10.02-24.97zm39.59-.17h5.02l10.63 24.97h-5.7l-2.26-5.56h-10.48l-2.27 5.56h-5.56l10.62-24.97zm5.73 14.59l-3.29-8.04-3.29 8.04h6.58zm18.62-14.42h5.46v19.83h12.36v4.96h-17.82v-24.79zm36.8 25.18c-3.34 0-5.95-.92-7.87-2.76-1.91-1.83-2.87-4.59-2.87-8.25v-14.16h5.46v14.02c0 2.03.47 3.56 1.42 4.58.95 1.03 2.26 1.54 3.93 1.54 1.67 0 2.99-.49 3.94-1.49.94-.99 1.41-2.48 1.41-4.46v-14.2h5.46v13.99c0 1.88-.26 3.53-.76 4.92-.51 1.39-1.24 2.55-2.18 3.49-.95.93-2.09 1.63-3.44 2.08-1.36.47-2.85.7-4.5.7zm21.82-25.18h18.67v4.85h-13.28v5.03h11.69v4.85h-11.69v5.2h13.46v4.86h-18.85v-24.79z"
  />
</svg>
`;
